src/spring-cloud/azext_spring_cloud/tanzu.py

```diff
@@ -148,8 +150,49 @@ def tanzu_app_deploy(cmd, client, resource_group, service, name, artifact_path,
     5. Send build result id to deployment
     '''
     deployment_name = _assert_deployment_exist_and_retrieve_name(cmd, client, resource_group, service, name)
-    # todo (qingyi)
-    build_result_id = ''
+    # get upload url
+    upload_url = None
+    relative_path = None
+    logger.warning("[1/3] Requesting for upload URL")
+    try:
+        response = client.build_service.get_upload_url(resource_group, service, None, None)
+        upload_url = response.upload_url
+        relative_path = response.relative_path
+    except (AttributeError, CloudError) as e:
+        raise CLIError("Failed to get a SAS URL to upload context. Error: {}".format(e.message))
+    # upload file
+    if not upload_url:
+        raise CLIError("Failed to get a SAS URL to upload context.")
+    account_name, endpoint_suffix, share_name, relative_name, sas_token = get_azure_files_info(upload_url)
+    logger.warning("[2/3] Uploading package to blob")
+    path = _get_tanzu_upload_local_file()
+    file_service = FileService(account_name, sas_token=sas_token, endpoint_suffix=endpoint_suffix)
+    file_service.create_file_from_path(share_name, None, relative_name, path)
+    if (target_module):
+        properties = models.BuildProperties("default-tanzu-builder", name, relative_path, {"BP_JVM_VERSION": "8.*", "BP_MAVEN_BUILT_MODULE": target_module})
+    else:
+        properties = models.BuildProperties("default-tanzu-builder", name, relative_path, None)
+    # create or update build
+    logger.warning("[3/3] Creating or Updating build '{}' (this operation can take a while to complete)".format(name))
+    build_result_id = None
+    try:
+        response = client.build_service.create_or_update_build(resource_group, service, name, properties, None, None)
+        build_result_id = response.TriggeredBuildResult.Id
+        build_result_list = response.TriggeredBuildResult.Id.split("/")
+        build_result_name = build_result_list[build_result_list.count()-1]

```

Comment:
`from msrestazure.tools import parse_resource_id` can be helpful

Reply:
> `from msrestazure.tools import parse_resource_id` can be helpful

OK, thanks~